src/ostree/ot-admin-builtin-finalize-staged.c

```diff
@@ -94,13 +84,11 @@ ot_admin_builtin_finalize_staged (int argc, char **argv, OstreeCommandInvocation
         return FALSE;
 
       /* We want to keep /boot open until the deployment is finalized during
-       * system shutdown, so block on SIGTERM under the assumption that it will
-       * be received when systemd stops the unit.
+       * system shutdown, so block until we get SIGTERM which systemd will send
+       * when the unit is stopped.  The below is basically the equivalent of
+       * `sleep infinity` because nothing will wake the event loop up.
        */
-      gboolean running = TRUE;
-      g_unix_signal_add (SIGTERM, sigterm_cb, &running);
-      g_print ("Waiting for SIGTERM\n");
-      while (running)
+      while (TRUE)
         g_main_context_iteration (NULL, TRUE);
```

Comment:
If the loop isn't doing anything, why not just use `pause()`? That's actually exactly what `sleep infinity` does. Seems silly to have glib setup an event loop that does nothing.

Reply:
Well, we're just optimizing a no-op at this point, but yes, done :smile: 
I didn't know about the `pause` system call actually.  It looks like it may be x86 only; glibc has a fallback that just uses `ppoll()` https://github.com/bminor/glibc/blob/02ca25fef2785974011e9c5beecc99b900b69fd7/sysdeps/unix/sysv/linux/pause.c#L31 with `NULL`s.

Reply:
I ran `strace` on `sleep infinity` and that's where it blocked. I think it is provided most everywhere on linux since it's in POSIX. `pause` is defined in most syscall tables in the kernel and there's a [generic implementation](https://github.com/torvalds/linux/blob/master/kernel/signal.c#L4602).